crates/shared/src/bad_token/cache.rs

```diff
@@ -53,38 +59,43 @@ impl CachingDetector {
         }
     }
 
-    fn insert_into_cache(&self, token: H160, quality: TokenQuality) {
+    async fn insert_into_cache(&self, token: H160, quality: TokenQuality) {
         self.cache
-            .lock()
-            .unwrap()
+            .write()
+            .await
             .insert(token, (Instant::now(), quality));
     }
 
-    fn insert_many_into_cache(&self, tokens: impl Iterator<Item = (H160, TokenQuality)>) {
-        let mut cache = self.cache.lock().unwrap();
+    async fn insert_many_into_cache(&self, tokens: impl Iterator<Item = (H160, TokenQuality)>) {
+        let mut cache = self.cache.write().await;
         let now = Instant::now();
-        for (token, quality) in tokens {
+        tokens.into_iter().for_each(|(token, quality)| {
             cache.insert(token, (now, quality));
-        }
+        });
```

Comment:
Nothing in particular, I find more idiomatic the zero-cost abstractions like `into_iter().for_each()`, different preferences :laughing:

Reply:
Doesn't closures provide performance overhead? Didn't know that classic loops have cost-related issues.

Reply:
Unless the compiler fucks up handrolled loops and iterators should result in the same assembly. In fact this particular example is exactly equivalent since `for item in list` is really just syntactic sugar for `list.into_iter().for_each()`.

Reply:
Classic loops doesn't have any cost-related issues. Actually both codes will compile to exactly the same or almost the same code.

What I meant is that `into_iter()` and `for_each()` are zero-cost abstraction functions, meaning that they don't incur more overhead than doing the iterator in the "classical" way (`for` in this case). And they have the advantage of providing potential easy chaining (like a `filter()`) in an easy way. And it is usually the preferred/most common way of doing iterations in Rust unless you want to have an early exit in the loop.

Reply:
> And they have the advantage of providing potential easy chaining (like a filter()) in an easy way. 

This is completely useless in this particular case. 

> And it is usually the preferred/most common way of doing iterations in Rust unless you want to have an early exit in the loop.

Ok, not a big deal. Just a bit surprised by that since Rust is not a purely functional programming language(where classic for loops are really anti-pattern). Still IMO, having fewer symbols is better.

Reply:
> This is completely useless in this particular case.

True, but you never know how the code will evolve in the future. That's the reason it is always preferred to do it like that when possible. Also adds consistency to the code base.